src/irr/asset/CGLTFLoader.cpp

```diff
@@ -34,16 +35,427 @@ namespace irr
+						if (uri.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							std::string_view uriBin = byteLength.get_string().value(); // TODO
+
+							const asset::IAssetLoader::SAssetLoadParams params;
+							auto buffer_bundle = assetManager->getAsset(rootAssetDirectory + uriBin.data(), params);
+							auto buffer = core::smart_refctd_ptr_static_cast<ICPUBuffer>(buffer_bundle.getContents().begin()[0]);
+						}
+						else
+							continue;
+					}
+				}
+
+				else if (key == "bufferViews")
+				{
+					for (auto& bufferView : value)
+					{
+						asset::SBufferBinding<ICPUBuffer> bufferBinding;
+
+						auto& buffer = bufferView.at_key("buffer");
+						if (buffer.error() != simdjson::error_code::NO_SUCH_FIELD)
+							buffer.get_uint64().value(); // TODO
+						else
+							continue;
+					}
+				}
+
+				/*
+					 Meshes are defined as arrays of primitives.
+					 Primitives correspond to the data required for GPU draw calls
+				*/
+
+				else if (key == "meshes")
+				{
+					for (auto& mesh : value)
+					{
+						auto& primitives = mesh.at_key("primitives");
+						if (primitives.error() != simdjson::error_code::NO_SUCH_FIELD)
+							for (auto& primitive : primitives)
+							{
+								auto& attributes = primitive.at_key("attributes");
+								if (attributes.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& position = attributes.at_key("POSITION");
+									auto& normal = attributes.at_key("NORMAL");
+									auto& tangent = attributes.at_key("TANGENT");
+									auto& texcoord0 = attributes.at_key("TEXCOORD_0");
+									auto& texcoord1 = attributes.at_key("TEXCOORD_1");
+									auto& color0 = attributes.at_key("COLOR_0");
+									auto& joint0 = attributes.at_key("JOINTS_0");
+									auto& weight0 = attributes.at_key("WEIGHTS_0");
+
+									// TODO
+								}
+								else
+									continue;
+							}
+						else
+							continue;
+					}
+				}
+
+				else if (key == "nodes")
+				{
+					for (auto& node : value)
+					{
+
+					}
+				}
+
+				/*
+					All large data for meshes, skins, and animations is stored in buffers and retrieved via accessors.
+					An accessor defines a method for retrieving data as typed arrays from within a bufferView.
+				*/
+
+				else if (key == "accessors")
+				{
+					for (auto& accessor : value)
+					{
+						auto& componentType = accessor.at_key("componentType");
+						if (componentType.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							auto& type = componentType.get_uint64().value();
+
+							switch (type)
+							{
+								case 5120: // BYTE
+								{
+
+								} break;
+
+								case 5121: // UNSIGNED_BYTE
+								{
+
+								} break;
+
+								case 5122: // SHORT
+								{
+
+								} break;
+
+								case 5123: // UNSIGNED_SHORT
+								{
+
+								} break;
+
+								case 5124: // UNSIGNED_INT
+								{
+
+								} break;
+
+								case 5125: // FLOAT
+								{
+
+								} break;
+
+								case 5126:
```

Comment:
`#define` the numerical literals and then undef them later

Reply:
or `enum`s

Reply:
solved